src/app/SERVICE/tasks.service.ts

```diff
@@ -1,61 +1,48 @@
 import { Injectable } from '@angular/core';
-import { Task } from '../MODELS/task.model';
-import { BehaviorSubject } from 'rxjs';
+import { Task } from '../models/task.model';
+import { BehaviorSubject, Observable, map } from 'rxjs';
+import { HttpClient } from '@angular/common/http';
+import { Router } from '@angular/router';
+import { Constants } from '../utils/constants';
 
 @Injectable({
   providedIn: 'root'
 })
 export class TasksService {
 
-  constructor() {
+  constructor(private http: HttpClient, private router: Router, private constants: Constants) {
   }
-  private tasks = new BehaviorSubject<Task[]>([
-    {
-      id: "8c410ecc-a204-43b6-9582-1fee127f858a",
-      title: "Grocery Shopping",
-      description: "Buy milk, bread, eggs, and fruits",
-      type: "Personal",
-      createdOn: new Date(),
-      status: "To Do",
-    },
-    {
-      id: "d468c77f-844c-4af6-b590-af2d8c7a2b9d",
-      title: "Finish Work Report",
-      description: "Complete the Q2 sales report for the manager",
-      type: "Work",
-      createdOn: new Date(2024, 6, 15),
-      status: "In Progress",
-    },
-    {
-      id: "e70626f2-4baf-4c3e-94ca-b0a9ce4c7a39",
-      title: "Pay Bills",
-      description: "Pay electricity and internet bills",
-      type: "Personal",
-      createdOn: new Date(2024, 6, 12),
-      status: "In Progress",
-    },
-  ])
 
-  availableTasks$ = this.tasks.asObservable()
-
-
-  createNewTask(task: Task) {
+  public getAllTasks():Observable<Task[]>{
+    return this.http.get<Task[]>(this.constants.tasksApiUrl)
+      .pipe(
+        map(data => { return data })
+      );
+  }
 
+  async createNewTask(task: Task) {
     if (task) {
-      const createdTask = this.tasks.getValue().slice()
-      createdTask.push(task)
-      this.tasks.next(createdTask)
-
+      await this.http.post(this.constants.tasksApiUrl, task).subscribe()
     } else {
-      console.log("no Tasks were Created")
+      return
     }
   }
-  deleteTask(deletedTask: Task) {
-    const updatedTasksArray = this.tasks.getValue().filter((task) => task !== deletedTask);
-    this.tasks.next(updatedTasksArray)
+  async deleteTask(deleteTaskId: string) {
+
+    this.http.delete<string>(this.constants.tasksApiUrl + deleteTaskId).subscribe(() => {
+    })
+  }
+  async taskDetails(tasksId: string) {
+   console.log(tasksId)
   }
-  taskDetails(id: string) {
-    const taskDetails = this.tasks.getValue().slice().filter((task) => task.id == id);
+  async updateTask(tasksNewData: Task) {
```

Comment:
Now that I've done some research, there's no point in using async await keywords when working with angular's http or anything asynchronous as it's very easy to do in Angular using observables, behaviorsubjects, subjects and subscriptions. An http call will always return an observable which you can either subscribe to from the component or using the async pipe in the components template